### ISSUE TITLE
String parameters of csoundSetOutput should be const

### ISSUE THREAD

**Issue by nwhetsell** [csound/csound#703]
I’m not sure if there are bigger ramifications of this but the [declaration of `csoundSetOption`](https://github.com/csound/csound/blob/0f0367bb38bde275771d1af3b231aa2b9156529a/include/csound.h#L849) —

```c
PUBLIC int csoundSetOption(CSOUND *csound, char *option);
``` 

should probably be

```c
PUBLIC int csoundSetOption(CSOUND *csound, const char *option);
``` 

This is probably also true of [`csoundCompileCsd`](https://github.com/csound/csound/blob/0f0367bb38bde275771d1af3b231aa2b9156529a/include/csound.h#L710), [`csoundSetOutput`](https://github.com/csound/csound/blob/0f0367bb38bde275771d1af3b231aa2b9156529a/include/csound.h#L903), and others.

**Comment by rorywalsh** [comment]
Agreed.

**Comment by vlazzarini** [comment]
ok will take a look.

**Comment by vlazzarini** [comment]
applied changes where relevant.

**Comment by nwhetsell** [comment]
The changes that closed this issue (commits eebeb9ae51b7dd9db16c70e6c95272b61fb61056 and 30e864dc27297a99e9157c4c702e4133195e73da) were mentioned on the CSOUND-DEV email list: https://listserv.heanet.ie/cgi-bin/wa?A2=CSOUND-DEV;f33936ad.1610.
